<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>768</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>微软雅黑</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>文法问题处理器 BNFProcessor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="80,30">
      <item>
       <widget class="QGroupBox" name="groupBox_edit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>操作台</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QTabWidget" name="tabWidget">
           <property name="currentIndex">
            <number>9</number>
           </property>
           <widget class="QWidget" name="tab_edit">
            <attribute name="title">
             <string>文法编辑</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <widget class="QLabel" name="label">
                   <property name="text">
                    <string>快捷编辑：</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_arrowChar">
                   <property name="text">
                    <string>→</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_orChar">
                   <property name="text">
                    <string>|</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_epsilonChar">
                   <property name="text">
                    <string>ɛ</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_process">
                   <property name="text">
                    <string>全部处理</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QPlainTextEdit" name="plainTextEdit_edit">
                 <property name="font">
                  <font>
                   <family>JetBrains Mono NL</family>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                 <property name="placeholderText">
                  <string>打开的文法会在此处显示，该窗口仅供预览，请勿编辑。</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_simplified">
            <attribute name="title">
             <string>文法化简</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QPlainTextEdit" name="plainTextEdit_simplified">
               <property name="font">
                <font>
                 <family>JetBrains Mono NL</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_check">
            <attribute name="title">
             <string>消除左递归</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QPlainTextEdit" name="plainTextEdit_leftRecursion">
               <property name="font">
                <font>
                 <family>JetBrains Mono NL</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab">
            <attribute name="title">
             <string>提取左公因子</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QPlainTextEdit" name="plainTextEdit_leftCommonFactor">
               <property name="font">
                <font>
                 <family>JetBrains Mono NL</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_first">
            <attribute name="title">
             <string>First集</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QTableWidget" name="tableWidget_firstSet">
               <property name="font">
                <font>
                 <family>JetBrains Mono</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <column>
                <property name="text">
                 <string>符号</string>
                </property>
                <property name="font">
                 <font>
                  <family>JetBrains Mono</family>
                  <pointsize>10</pointsize>
                 </font>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>集合</string>
                </property>
                <property name="font">
                 <font>
                  <family>JetBrains Mono</family>
                  <pointsize>10</pointsize>
                 </font>
                </property>
               </column>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_2">
            <attribute name="title">
             <string>Follow集</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <item>
              <widget class="QTableWidget" name="tableWidget_followSet">
               <property name="font">
                <font>
                 <family>JetBrains Mono</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <column>
                <property name="text">
                 <string>符号</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>集合</string>
                </property>
               </column>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_LL1">
            <attribute name="title">
             <string>LL(1)分析表</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_12">
             <item>
              <widget class="QTableWidget" name="tableWidget_LL1">
               <property name="font">
                <font>
                 <family>JetBrains Mono NL</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_CST">
            <attribute name="title">
             <string>语法分析</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_14">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_13">
               <item>
                <widget class="QPlainTextEdit" name="plainTextEdit_CST">
                 <property name="font">
                  <font>
                   <family>JetBrains Mono NL</family>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                 <property name="placeholderText">
                  <string>在此处输入需要语法分析的源程序</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <item>
                  <widget class="QLabel" name="label_2">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>源程序语言：</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QComboBox" name="comboBox_language">
                   <item>
                    <property name="text">
                     <string>TINY</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>MiniC</string>
                    </property>
                   </item>
                  </widget>
                 </item>
                 <item>
                  <spacer name="verticalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>40</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_3">
            <attribute name="title">
             <string>分析树</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_19">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_10">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_20">
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>当前语言：</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_language">
                   <property name="text">
                    <string>TINY</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_AST">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>生成语法树</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QTreeWidget" name="treeWidget_CST">
                 <property name="font">
                  <font>
                   <family>JetBrains Mono NL</family>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                 <column>
                  <property name="text">
                   <string notr="true">1</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_4">
            <attribute name="title">
             <string>语法树</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_21">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_9">
               <item>
                <widget class="QTreeWidget" name="treeWidget_AST">
                 <property name="font">
                  <font>
                   <family>JetBrains Mono</family>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                 <column>
                  <property name="text">
                   <string notr="true">1</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="40,80">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,5">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QGroupBox" name="groupBox_4">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>147</width>
                <height>117</height>
               </size>
              </property>
              <property name="title">
               <string>文法</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_16">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QPushButton" name="pushButton_open">
                   <property name="text">
                    <string>打开</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_save">
                   <property name="text">
                    <string>保存</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_5">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>147</width>
                <height>117</height>
               </size>
              </property>
              <property name="title">
               <string>操作1-2</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_17">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <widget class="QPushButton" name="pushButton_simplify">
                   <property name="text">
                    <string>文法化简</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_eliminateLeftRecursion">
                   <property name="text">
                    <string>消除左递归</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>183</width>
              <height>225</height>
             </size>
            </property>
            <property name="title">
             <string>操作3-6</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_18">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_8">
               <item>
                <widget class="QPushButton" name="pushButton_eliminateLeftCommonFactor">
                 <property name="text">
                  <string>提取左公因子</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_set">
                 <property name="text">
                  <string>集合求解</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_LL1">
                 <property name="text">
                  <string>生成LL(1)分析表</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_CST">
                 <property name="text">
                  <string>语法分析</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushButton_clearAll">
                 <property name="text">
                  <string>复位</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_console">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>341</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>控制台</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QPlainTextEdit" name="plainTextEdit_console">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>315</width>
                 <height>332</height>
                </size>
               </property>
               <property name="font">
                <font>
                 <family>Courier New</family>
                 <pointsize>9</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_clearConsole">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_edit">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="action_simplify"/>
    <addaction name="action_check"/>
    <addaction name="action_set"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_edit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_open">
   <property name="text">
    <string>打开(&amp;O)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>保存(&amp;S)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_simplify">
   <property name="text">
    <string>文法化简</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_check">
   <property name="text">
    <string>文法检查</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+C</string>
   </property>
  </action>
  <action name="action_set">
   <property name="text">
    <string>集合求解</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+S</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>退出(&amp;Q)</string>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Q</string>
   </property>
  </action>
  <action name="actionAbout_me">
   <property name="text">
    <string>About me.😋</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
